### DISCUSSION TITLE
HMR doesn't work when external react and react-dom.

### DISCUSSION THREAD

**Discussion by MrKou47** [vercel/next.js#30252] [Help] [answered]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

16.11.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

No. But you can check reproduce link below.

### Describe the Bug

I have an application which required `window.React` and `window.ReactDOM`, so i modify the webpack externals field in *next.config.js* like this:
```js
// next.config.js
/** @type {import('next').NextConfig} */
module.exports = {
  reactStrictMode: true,
  webpack(config, { isServer }) {
    if (!isServer) {
      config.externals = {
        next: 'next',
        react: 'React',
        'react-dom': 'ReactDOM'
      }
    }
    return config;
  }
}
```
But when i do this, the page can't hot reload when editing code. And there is no warning or error in the Chrome console panel. Only when i manually refresh the page, the latest code can work as well.

### Expected Behavior

HMR works as usual.

### To Reproduce

https://github.com/MrKou47/nextjs-hmr-notwork-demo

**Comment by edomonticelli** [comment, marked as answer]
I can confirm the issue

@MrKou47 the { isServer } made my day thanks :)

**Comment by ndreckshage** [comment]
HMR works with this approach if you switch to the development version of react & react-dom, to note. Your example uses the production versions.
